### DISCUSSION TITLE
Feature request: rename `imap` to `importmap`

### DISCUSSION THREAD

**Discussion by nnmrts** [jurassiscripts/velociraptor#100] [Ideas]
This would have several positive implications:
- Deno itself uses `--importmap` not `--imap`. Users, especially new ones, probably expect vr to use the same names for options like these.
- [denon uses `importmap`](https://github.com/denosaurs/denon#import-map) as well. This was the main motivation for this feature request, as changing this option to `importmap` in vr would make the config formats of both "compatible", in the sense that one doesn't have to add both `imap` and `importmap` options to their config files. Note that it's still not really possible to use one config file for both vr and denon due to denon using the `scripts.config.<EXT>` format for dynamic config files for some reason. I'll create PR for this at the denon repo.
- It makes more sense. I understand abbreviating stuff as a concept, don't get me wrong. I don't think this the right place to do it though. The term [`imap` is not unambiguous](https://en.wikipedia.org/wiki/Internet_Message_Access_Protocol) and also not widely used elsewhere across the Deno community, as far as I know.

I'll create two PRs for this, one breaking and one non-breaking, by adding a deprecation notice when using `imap`.

**Comment by nnmrts** [comment]
The two PRs:
breaking: https://github.com/jurassiscripts/velociraptor/pull/101
non-breaking: https://github.com/jurassiscripts/velociraptor/pull/102

**Reply by nnmrts** [reply]
PR for the docs: https://github.com/jurassiscripts/velociraptor-dot-run/pull/2

**Comment by umbopepato** [comment]
Hey @nnmrts, I completely agree and would like to take the occasion to uniform the option names to Deno's ones (#95  as well). I would probably keep the existing names as additional aliases for backward compatibility. Would you mind adding the `config` alias to the non-breaking PR and switch the base to `develop`?

**Reply by nnmrts** [reply]
I have to admit a mistake though, Deno uses `import-map`, with a dash. This should translate to a property called `importMap`, not `importmap`. So actually both denon and vr don't really follow Deno's way of doing things. Your thoughts on this?

**Reply by nnmrts** [reply]
Pushed a fix regarding this to the PR but I think I should wait on your thoughts on this before going forward...sorry for the mistake, don't really know how I missed that.

Seems like Deno changed it from `importmap` to `import-map` in 1.5.0, but `importmap` still works I guess...

Compare:
https://deno.land/manual@v1.4.6/linking_to_external_code/import_maps
with
https://deno.land/manual@v1.5.0/linking_to_external_code/import_maps
